spring/spring4/spring4-remote/pom.xml

```diff
@@ -22,6 +22,12 @@
         <dependency>
             <groupId>org.springframework</groupId>
             <artifactId>spring-context</artifactId>
+            <scope>provided</scope>
```

Comment:
Marking everything as ```provided``` has a side effect of letting users on their own when it comes to dependency management. Example, if I add to my pom
```
<dependency>
    <groupId>org.infinispan</groupId>
    <artifactId>infinispan-spring4-remote</artifactId>
    <version>9.0.0-SNAPSHOT</version>
 </dependency>
```
and print the tree I get only:
```
[INFO] \- org.infinispan:infinispan-spring4-remote:jar:9.0.0-SNAPSHOT:compile
[INFO]    \- org.infinispan:infinispan-spring4-common:jar:9.0.0-SNAPSHOT:compile
```

Which means I need to figure out myself the spring and and hot rod client dependencies (I guess there is a handful of them), which normally involves playing a dependency bingo game. :smile:

What is the motivation of not using dependencies with the right scope?